services/beaconchain/src/main/java/tech/pegasys/artemis/services/beaconchain/BeaconChainController.java

```diff
@@ -108,6 +110,9 @@ public BeaconChainController(
     this.metricsSystem = metricsSystem;
     this.setupInitialState =
         config.getDepositMode().equals(DEPOSIT_TEST) || config.getStartState() != null;
+    this.slotEventsChannelPublisher = eventChannels.getPublisher(SlotEventsChannel.class);
+
+    eventChannels.subscribe(TimeTickChannel.class, this);
```

Comment:
Probably shouldn't be subscribing in the constructor but do it in the `init` method instead.

Reply:
Definitely shouldn't subscribe here.  We're getting:
```
2020-03-25 10:11:39.682+10:00 | TimeTickChannel-0 | FATAL | teku-status-log | PLEASE FIX OR REPORT | Unexpected exception thrown for java.lang.NullPointerException: event 'interface tech.pegasys.artemis.util.time.channels.TimeTickChannel.onTick' in handler 'tech.pegasys.artemis.services.beaconchain.BeaconChainController'
java.lang.NullPointerException: null
	at tech.pegasys.artemis.services.beaconchain.BeaconChainController.onTick(BeaconChainController.java:335) ~[classes/:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
	at tech.pegasys.artemis.events.DirectEventDeliverer.executeMethod(DirectEventDeliverer.java:67) ~[classes/:?]
	at tech.pegasys.artemis.events.DirectEventDeliverer.deliverTo(DirectEventDeliverer.java:54) ~[classes/:?]
	at tech.pegasys.artemis.events.AsyncEventDeliverer.lambda$deliverTo$0(AsyncEventDeliverer.java:60) ~[classes/:?]
	at tech.pegasys.artemis.events.AsyncEventDeliverer$QueueReader.deliverNextEvent(AsyncEventDeliverer.java:111) [classes/:?]
	at tech.pegasys.artemis.events.AsyncEventDeliverer$QueueReader.run(AsyncEventDeliverer.java:103) [classes/:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
```

So suspect we should only subscribe to tick events *after* we've called all the `init*` methods.

Reply:
Makes sense. My bad.

Reply:
Done.